src/wallet/wallet.h

```diff
@@ -1206,6 +1206,18 @@ class CWallet final : public CCryptoKeyStore, public CValidationInterface
     /** overwrite all flags by the given uint64_t
        returns false if unknown, non-tolerable flags are present */
     bool SetWalletFlags(uint64_t overwriteFlags, bool memOnly);
+
+    /** Returns a bracketed wallet name for displaying in logs, will return [default wallet] if the wallet has no name */
+    const std::string GetDisplayName() const {
+        return "[" + (GetName().length() == 0 ? "default wallet" : GetName()) + "]";
+    };
+
+    /** Prepends the wallet name in logging output to ease debugging in multi-wallet use cases */
+    template<typename... Params>
+    void WalletLogPrintf(std::string fmt, Params... parameters) const {
+        LogPrintf((GetDisplayName() + " " + fmt).c_str(), parameters...);
```

Comment:
sanitize inputs please :)

```
/home/ubuntu/bitcoin
→ bcli createwallet "a%ib"
{
  "name": "a%ib",
  "warning": ""
}
tail -30 ../.bitcoin/regtest/debug.log
2018-07-27T14:36:38Z tor: Not connected to Tor control port 127.0.0.1:9051, trying to reconnect
2018-07-27T14:36:40Z tor: Error connecting to Tor control socket
2018-07-27T14:36:40Z tor: Not connected to Tor control port 127.0.0.1:9051, trying to reconnect
2018-07-27T14:36:44Z tor: Error connecting to Tor control socket
2018-07-27T14:36:44Z tor: Not connected to Tor control port 127.0.0.1:9051, trying to reconnect
2018-07-27T14:36:45Z Received a POST request for / from 127.0.0.1:59554
2018-07-27T14:36:45Z ThreadRPCServer method=createwallet user=bitcoinrpc
2018-07-27T14:36:45Z Using BerkeleyDB version Berkeley DB 4.8.30: (April  9, 2010)
2018-07-27T14:36:45Z Using wallet wallet.dat
2018-07-27T14:36:45Z BerkeleyEnvironment::Open: LogDir=/home/ubuntu/.bitcoin/regtest/wallets/a%ib/database ErrorFile=/home/ubuntu/.bitcoin/regtest/wallets/a%ib/db.log
2018-07-27T14:36:45Z init message: Loading wallet a%ib...
2018-07-27T14:36:45Z Error "tinyformat: Too many conversion specifiers in format string" while formatting log message: [a%ib] nFileVersion = %d
2018-07-27T14:36:45Z Error "tinyformat: Too many conversion specifiers in format string" while formatting log message: [a%ib] Keys: %u plaintext, %u encrypted, %u w/ metadata, %u total. Unknown wallet records: %u
2018-07-27T14:36:45Z Error "tinyformat: Too many conversion specifiers in format string" while formatting log message: [a%ib] Performing wallet upgrade to %i
2018-07-27T14:36:46Z Error "tinyformat: Too many conversion specifiers in format string" while formatting log message: [a%ib] keypool added %d keys (%d internal), size=%u (%u internal)
2018-07-27T14:36:46Z Error "tinyformat: Too many conversion specifiers in format string" while formatting log message: [a%ib] wallet      %15dms
2018-07-27T14:36:46Z Error "tinyformat: Too many conversion specifiers in format string" while formatting log message: [a%ib] setKeyPool.size() = %u
2018-07-27T14:36:46Z Error "tinyformat: Too many conversion specifiers in format string" while formatting log message: [a%ib] mapWallet.size() = %u
2018-07-27T14:36:46Z Error "tinyformat: Too many conversion specifiers in format string" while formatting log message: [a%ib] mapAddressBook.size() = %u
2018-07-27T14:36:49Z Flushing wallet.dat
2018-07-27T14:36:49Z Flushed wallet.dat 4ms
2018-07-27T14:36:49Z tor: Error connecting to Tor control socket
2018-07-27T14:36:49Z tor: Not connected to Tor control port 127.0.0.1:9051, trying to reconnect
2018-07-27T14:36:56Z tor: Error connecting to Tor control socket
2018-07-27T14:36:56Z tor: Not connected to Tor control port 127.0.0.1:9051, trying to reconnect
2018-07-27T14:37:08Z tor: Error connecting to Tor control socket
2018-07-27T14:37:08Z tor: Not connected to Tor control port 127.0.0.1:9051, trying to reconnect
2018-07-27T14:37:25Z tor: Error connecting to Tor control socket
2018-07-27T14:37:25Z tor: Not connected to Tor control port 127.0.0.1:9051, trying to reconnect
2018-07-27T14:37:36Z Adding fixed seed nodes as DNS doesn't seem to be available.
```

Reply:
This would probably conflict with unicode support in wallet file names, no?

Reply:
Should probably escape or quote instead.

Reply:
@MarcoFalke I definitely agree that escape or quote would be better than sanitize (which simply drops characters), however the only existing escape functions I found were in leveldb, qt, and univalue. Any advice before I try my hand at adding one?

Reply:
Rather than adding an escape fn another option is to just 

>LogPrintf("%s " + fmt, GetDisplayName(), parameters...)

Reply:
Good point!